source/tutorial.rst

```diff
@@ -57,8 +57,21 @@ Control structures
 Functions
 ^^^^^^^^^^^^
 
-``Work in progress!``
+The way to define functions is similar as it is in other languages.
+
+Note that the return type must be defined if you intend to return anything from the function. If the return type is not specified, the function is assumed to be of type ``void``, so no ``return`` statement is allowed inside the function.
+
+Also note that the arguments passed into any function are passed by reference, meaning that the changes made to the parameters inside the function will affect the originals.
+
+.. code:: kotlin
+    
+    fun func1(number x): number {
+        return x + 1;
+    }
 
+    fun func2(Stack<number> stack) {  // function assumed to be void as no return type is specified
```

Comment:
Also I just tried this code in our language and it doesn't compile 😅  can we use the Manim DSL code comment symbol which is a `#`?

Reply:
I also realise it wouldn't compile regardless, since right now the program needs at least one statement to be syntactically correct. But I think it would be good to use the syntax we have defined

Reply:
oh sorry the comment here was just meant to show the audience (kind of copied the style from the rest of the page). With the empty body, I thought I changed it so we can have empty function body? Or did we change it again

Reply:
Oh I see, so the rest of the page has `//`? I might change that on the next PR

Reply:
Ah no no, you did change it. I just meant that these code snippets don't necessarily _have_ to be files that would compile, since our current parser grammar requires at least one global statement after the functions have (optionally) been defined. Sorry for the confusion

Reply:
I'll double check the current syntax about function body

Reply:
> Ah no no, you did change it. I just meant that these code snippets don't necessarily _have_ to be files that would compile, since our current parser grammar requires at least one global statement after the functions have (optionally) been defined. Sorry for the confusion

oh I see yeah these are just examples of the function syntax but I can add statements in?

Reply:
You could add a statement just below like `let y = func1(5);` if you wanted? Just to make it crystal clear to the user